function_calling/src/function_calling.py

```diff
@@ -0,0 +1,198 @@
+import os
```

Comment:
I would add to the readme a link to this file of examples if you haven't already. If you already did, please ignore.

Reply:
I would define here all the classes and functions of the notebook that haven't been defined.